routing/pathfind.go

```diff
@@ -353,22 +354,25 @@ func getMaxOutgoingAmt(node route.Vertex, outgoingChan *uint64,
 		// fail the pathfinding early.
 		if !ok {
 			max = lnwire.MaxMilliSatoshi
+			total = lnwire.MaxMilliSatoshi
```

Comment:
Even for the `max` setting above, I think this should instead be the _capacity_ of that channel, not the largest possible mSAT amount that can be expressed.

Reply:
I think we're saved here atm by the checks to max_htlc, otherwise we'd keep trying amounts that can't ever be carried by the target channel.

Reply:
Iirc this clause is to deal with the case where we query bandwidth hints and a split second later the channel becomes active and shows up in the graph here.

This is just a pre-check. The worst that can happen is that we pathfind and can't find no route. However for the newly added insufficient local balance failure, it would split one more time.

Using capacity makes this all slightly less likely to happen. Capacity is easily accessible, so I fixed that.